alpha/lib/model/Partner.php

```diff
@@ -2454,4 +2432,48 @@ public function setAllowedEmailDomainsForAdmins($v)
 	{
 		return $this->putInCustomData(self::ALLOWED_EMAIL_DOMAINS_FOR_ADMINS, $v);
 	}
+	
+	public function getSupportAccessAllowedUntil()
+	{
+		return $this->getFromCustomData(self::SUPPORT_ACCESS_ALLOWED_UNTIL);
+	}
+	
+	public function setSupportAccessAllowedUntil($v)
+	{
+		return $this->putInCustomData(self::SUPPORT_ACCESS_ALLOWED_UNTIL, $v);
+	}
+	
+	public function getSupportAccessMode()
+	{
+		return $this->getFromCustomData(self::SUPPORT_ACCESS_MODE, null, SupportAccessMode::ALLWAYS_ALLOWED);
+	}
+	
+	public function setSupportAccessMode($v)
+	{
+		return $this->putInCustomData(self::SUPPORT_ACCESS_MODE, $v);
+	}
+	
+	public function isSupportAccessAllowed()
+	{
+		$shouldEnforceSupportAccess = kConf::get('enforce_support_access', kConfMapNames::SECURITY, true);
+		if(!$shouldEnforceSupportAccess || $this->getSupportAccessMode() == SupportAccessMode::ALLWAYS_ALLOWED)
+		{
+			return true;
+		}
+		
+		$allowedUntil = $this->getSupportAccessAllowedUntil();
+		$allowedUntil = $allowedUntil ? $allowedUntil : (time() - 3600);
+		return (time() < $allowedUntil);
```

Comment:
in the case you set allowedUntil= time() - 3600 this condition in line 2466 will always be false, is this intended?